File: Exercício Incremental/04/Endereco.java
import java.util.ArrayList;

public class Endereco {
  private String rua;
  private int numero;
  private String cep;
  private String estado;
  private String cidade;
  // Siglas de estados permitidos
  private ArrayList<String> estadosPermitidos = new ArrayList<String>() {{
    add("AC"); add("AL"); add("AP"); add("AM"); add("BA"); add("CE"); add("DF");
    add("ES"); add("GO"); add("MA"); add("MT"); add("MS"); add("MG"); add("PA");
    add("PB"); add("PR"); add("PE"); add("PI"); add("RJ"); add("RN"); add("RS");
    add("RO"); add("RR"); add("SC"); add("SP"); add("SE"); add("TO");
  }};

  public Endereco(String rua, int numero, String cep, String estado, String cidade) {
    // Validar estado
    if(!(estadosPermitidos.contains(estado.toUpperCase()))) {
      throw new IllegalArgumentException("A sigla do estado informado é inválida.");
    }
    this.estado = estado.toUpperCase();
    this.rua = rua;
    this.numero = numero;
    this.cep = cep;
    this.cidade = cidade;
  }

  // Caso não seja informado estado e cidade
  public Endereco(String rua, int numero, String cep) {
    this(rua, numero, cep, "BA", "Salvador");
  }

  // GETs
  public String getRua() {
    return this.rua;
  }

  public int getNumero() {
    return this.numero;
  }

  public String getCep() {
    return this.cep;
  }

  public String getEstado() {
    return this.estado;
  }

  public String getCidade() {
    return this.cidade;
  }

  // SETs
  public void setRua(String novaRua) {
    this.rua = novaRua;
  }

  public void setNumero(int novoNumero) {
    this.numero = novoNumero;
  }

  public void setCep(String novoCep) {
    this.cep = novoCep;
  }

  public void setEstado(String novoEstado) {
    this.estado = novoEstado;
  }

  public void setCidade(String novaCidade) {
    this.cidade = novaCidade;
  }
}

File: Trabalho Final/endereco.py
class Endereco:
  # Siglas dos estados permitidos
  estados_permitidos = [
    "AC", "AL", "AP", "AM", "BA", "CE", "DF", "ES", "GO", 
    "MA", "MT", "MS", "MG", "PA", "PB", "PR", "PE", "PI", 
    "RJ", "RN", "RS", "RO", "RR", "SC", "SP", "SE", "TO"
  ]

  def __init__(self, rua, numero, cep, estado="BA", cidade="Salvador"):
    if estado.upper() not in self.estados_permitidos:
      raise ValueError("A sigla do estado informado é inválida.")
    self.rua = rua
    self.numero = numero
    self.cep = cep
    self.estado = estado.upper()
    self.cidade = cidade

  def __str__(self):
    return (
      f"Rua/Avenida: {self.rua}, Número: {self.numero}\n" +
      f"CEP: {self.cep}, Estado: {self.estado}, Cidade: {self.cidade}"
    )
  
  # GETs
  def get_rua(self):
    return self.rua

  def get_numero(self):
    return self.numero

  def get_cep(self):
    return self.cep

  def get_estado(self):
    return self.estado

  def get_cidade(self):
    return self.cidade

  # SETs
  def set_rua(self, nova_rua):
    self.rua = nova_rua

  def set_numero(self, novo_numero):
    self.numero = novo_numero

  def set_cep(self, novo_cep):
    self.cep = novo_cep

  def set_estado(self, novo_estado):
    if novo_estado.upper() not in self.estados_permitidos:
      raise ValueError("A sigla do estado informado é inválida.")
    self.estado = novo_estado.upper()

  def set_cidade(self, nova_cidade):
    self.cidade = nova_cidade
